### PR TITLE
DB properties layout and save order config changed

### LINKED ISSUE #957: Save actions: better field selection
At "Save sort order" in the same dialog, one can also select a field. I would have expected the same choice for the selection of the fields to apply the save action. Is that possible? Reason: I am too lazy to input the field name manually.

grabbed_20160313-212741

### PR DESCRIPTION
**Pull request by Siedlerchr**
- [x] Change in CHANGELOG.md described?
- [x] Changes in pull request outlined? (What, why, ...)
- [x] Tests green?

Fixes  #957  and #958 

Fix height of the dialog and add spacing between OK/Cancel Button and  border
Changed JComboxBox to editable and removed the unnecessary textfields
Added JCombobox for selecting the fields to apply the selected formatter to
Changed position of Delete Button
Removed IndexOutOfBound Exception when no item is selected

Edit// I just see this PR is #999 ;)

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsListModel.java [hunk 1]
```diff
@@ -27,20 +27,21 @@ public void addSaveAction(FieldFormatterCleanup action) {
...
         }
     }
 
     public void removeAtIndex(int index) {
-        if (index < 0 || index > saveActions.size()) {
-            throw new IndexOutOfBoundsException("Index must be within 0 and " + saveActions.size() + " but was " + index);
-        }
 
-        FieldFormatterCleanup action = saveActions.remove(index);
+        if ((index >= 0) && (index < saveActions.size())) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsListModel.java]: Shouldn't the error message just be logged or is it often the case that the index parameter is given "out of bounds"?

**Comment by koppor** [comment]
Is it possible to write a simple UI test for it? :innocent:

Refs #507

File: src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsListModel.java (hunk 1 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsListModel.java]: It is always the case when you click the button and nothing is selected in the ListBox or you just deleted the last entry.
So I thought: Nothing selected/nothing to remove -> Do nothing

**Comment by Siedlerchr** [comment]
UI-Test, I will try to write one

**Comment by Siedlerchr** [comment]
Added GUI Test und fixed Checkbox label not displaying correclty

**Comment by tobiasdiez** [comment]
The code looks good, so I give my :+1: for merging if you add a nice looking screenshot :smiley:

**Comment by Siedlerchr** [comment]
@tobiasdiez  Here you are:
dbproperties

File: src/main/java/net/sf/jabref/gui/dbproperties/DatabasePropertiesDialog.java [hunk 2]
```diff
@@ -105,10 +105,11 @@ private void init(JFrame parent) {
         browseFile.addActionListener(BrowseAction.buildForDir(parent, fileDir));
         browseFileIndv.addActionListener(BrowseAction.buildForDir(parent, fileDirIndv));
 
-        setupSortOrderConfiguration();
 
-        FormBuilder builder = FormBuilder.create().layout(new FormLayout("left:pref, 4dlu, left:pref, 4dlu, pref:grow",
-                "pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 140dlu, pref,"));
+        setupSortOrderConfiguration();
+        FormLayout form = new FormLayout("left:pref, 4dlu, left:pref, 4dlu, pref:grow",
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/dbproperties/DatabasePropertiesDialog.java]: Change the grow to the third column? Now, the browse button is resized. better if the text field is resized.

File: src/main/java/net/sf/jabref/gui/dbproperties/DatabasePropertiesDialog.java [hunk 3]
```diff
@@ -105,10 +105,11 @@ private void init(JFrame parent) {
...
+                "pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 2dlu, pref, 160dlu, pref,");
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/dbproperties/DatabasePropertiesDialog.java]: Add fill/grow to the row corresponding to the save action panel. Makes sense that it is expanded if the dialog is resized.

**Comment by Siedlerchr** [comment]
@oscargus  
I tried to fix this, but I could not get the damn JList resized horizontally or vertically. I tried around a lot. 
The code for this is located in `FieldFormattersCleanUpPanel`

From this one it sounded very simple:

http://alvinalexander.com/java/edu/jgoodies-formlayout-grow-fill/3-fixing-jgoodies-formlayout-sizing.shtml
dbpropertiesresized

**Comment by oscargus** [comment]
My code did everything except horizontal sizing (and I also tried a
lot...), so check there for possible hints. I think that when spanning
multiple columns using a fourth alignment argument to .xyw is one thing.

**Comment by oscargus** [comment]
@Siedlerchr : I think it is good to go, so please rebase and it can be merged.

(I reserve the right to fix/change(?) the scaling at some stage. :-))

**Comment by Siedlerchr** [comment]
@oscargus  Okay, otherwise I would have looked at the GUI resizing in more detail the next days

**Comment by oscargus** [comment]
@Siedlerchr Up to you. I just wasn't sure what the status of the PR was and was thinking of merging it (if it was possible). Feel free to look at the resizing. :+1:

**Comment by Siedlerchr** [comment]
@oscargus  As I just rebased it on the master I would suggest merging it for once now, so that the biggest changes (e.g. the Comboxes) are into it.
The resizing are only some minor things.  I will try to experiment with them in the next days and release it then as a new pr.

**Pull request merged**